src/librustc_middle/ty/context.rs

```diff
@@ -926,12 +928,11 @@ pub struct GlobalCtxt<'tcx> {
     pub(crate) untracked_crate: &'tcx hir::Crate<'tcx>,
     pub(crate) definitions: &'tcx Definitions,
 
-    /// A map from `DefPathHash` -> `DefId`. Includes `DefId`s from the local crate
-    /// as well as all upstream crates. Only populated in incremental mode.
-    pub def_path_hash_to_def_id: Option<FxHashMap<DefPathHash, DefId>>,
-
     pub queries: query::Queries<'tcx>,
 
+    pub local_def_path_hash_to_def_id: Option<FxHashMap<DefPathHash, DefId>>,
```

Comment:
```suggestion
    pub local_def_path_hash_to_def_id: FxHashMap<DefPathHash, LocalDefId>,
```
Looks like it would be simpler to just use an empty map instead of an option.
(Also def id -> local def id.)

Reply:
I used an `Option` to catch any code that tries to read from this when incremental compilation is disabled.

Reply:
I ended up getting rid of the `Option` when I moved this to `on_disk_cache`